src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntity.php

```diff
@@ -30,6 +30,7 @@ class UniqueEntity extends Constraint
     public $em = null;
     public $entityClass = null;
     public $repositoryMethod = 'findBy';
+    public $isEntityToUpdateMethod = null;
```

Comment:
maybe `isEqualToEntityMethod`? Basically it is what the validator asks to throw the violation or not.

Reply:
DTO can't be equal to an entity

Reply:
It would make no sense to call it if an entity is created: if an entity exists in database with fields of same value it cannot be the one which comes from the DTO as it isn't in the database yet.

Reply:
When we are creating a new entity through a DTO, we also need to call to the comparison method to check the equality.

Take the example below, on create the `$this->authorId` is `null`, on update the `$this->authorId` is the entity ID that is being modified.
```php
/**
 * @UniqueEntity(fields={"email"}, entityClass="App\Entity\Author", isEqualToMethod="isEqualTo")
 */
class AuthorDTO
{
    private $authorId;
    private $email;

    // ...

    public function isEqualTo($entity)
    {
        return $entity->getId() === $this->authorId;
    }
}
```
Results:
* On create always return `false`, so throw the violation.
* On update depends of the equality, if `true` pass, else throw the violation.

The validator doesn't know nothing about the current event, so the method must be called always.

Reply:
I also think something like `isEqualToEntityMethod` would make more sense.